### DISCUSSION TITLE
sOnOff POW 2 Weekly and Monthly Data

### DISCUSSION THREAD

**Discussion by LurchiStromberg** [arendst/Tasmota#16992] [General]
Hi!
It would be great, if the sOnOff POW2 would be able to show weekly and monthly values in the Webinterface.
I looked at Google but found nothing.

Is it really, that nobody is interessted in those Values?
Daily, Yesterday and Yearly is great, but i also like to see the weekly and monthly kW/h.

Any Chance to add this in next Firmware?

**Comment by barbudor** [comment]
Most people interested I gathering in depth statistics are doing that off-device in a home automation backend where you can easily store data from multiple device, compare, compute, graph as you want

On Tasmota, you can extended features using Scripting Language or Berry language including adding stuff in the web GUI.

**Reply by sfromis** [reply]
Using many of those energy monitoring devices, I'm keeping historical data in my backend, and on-device weekly/monthly values would just not be something I'd be be very interested in. Tasmota is not really designed for stand-alone operation, more to work with a backend platform, with the "M" in the name meaning "MQTT".

**Reply by LurchiStromberg** [reply]
I have seen somewhere (i used Google but never found it again) a (maybe) custom Webinterface in Tasmote what showes the Values for daily / monthly / yearly

Sure people who are able to flash Tasmota should be more into programming and coding.
But i am not.
Its easy for me to do the Hardware Site but at coding i am a Noob.

I think for a Dev its an easy Thing to include such Features at Tasmota.
And if you ask me, its more usable to have the Values directy at the Device as at a Platform like FHEM / Iobroker or else....

If its as easy to include as you said, please give me a hint how to add it at scripting on the Device.
